### ISSUE TITLE
Formulation : Tests 1.1.1, 1.1.2, 1.1.3 : Images avec attributs `alt`

### ISSUE THREAD

**Issue by innoticFR** [DISIC/RGAA#62]
Les balises non décoratives `area`, `img` et `input[type='image']` doivent posséder l'attribut `alt` non vide.

La [technique ARIA10 ](https://www.w3.org/TR/WCAG20-TECHS/ARIA10.html) (`aria-labelledby`) n'est normalement pas applicable aux balises munies d'un attribut `alt` pour fournir une alternative textuelle
> 1. Examine each element where the `aria-labelledby` attribute is present and **the element does not support the `alt` attribute**.

Il faut donc retirer `aria-labelledby` (et en toute logique `aria-label`) des alternatives textuelles *"suffisantes"* dans la note du glossaire "Alternative textuelle (image)", même si celles-ci entrent effectivement dans l'algorithme du nom accessible pour les technologies d'assistance, abordé au critère 1.3. 

Voir notamment [Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/GL/wiki/Using_aria-labelledby_to_provide_a_text_alternative_for_non-text_content)
> Note: When using `aria-labelledby` attribute on an `img` element, the absence of an `alt` attribute will cause validation failures for HTML 4 and earlier and XHTML. If `alt` is provided, it must contain the same text as the `aria-labelledby` target. **Do not use `alt=""`, which would cause some assistive technology to ignore the image completely**.

L'objectif ici est de fournir une alternative textuelle à tous les utilisateurs, pas seulement aux utilisateurs de technologies d'assistance. En outre, une balise `<img>` sans attribut `alt` n'a pas de rôle ARIA `img` implicite.

L'utilisation des attributs `aria-label` et `aria-labelledby` reste toutefois possible, en complément à l'attribut `alt`, sur ces tags afin de fournir un texte adapté aux technologies d'assistance (ex: "Revenir à la page", plutôt que "fermer le popup")

**Comment by alixraison** [comment]
Le fait que les éléments area, img et input[type='image'] pouvant disposer d’une alternative au moyen d’un attribut alt, les autres possibilités d’alternative (aria-labelledby, aria-label) ne devraient pas être considérées comme valides selon la technique ARIA10 qui signale dans sa procédure de test : “Examine each element where the aria-labelledby attribute is present and the element does not support the alt attribute”.

Or selon la technique F65 (https://www.w3.org/WAI/WCAG21/Techniques/failures/F65) 
l’ensemble des autres possibilités techniques (aria-labelledby, aria-label et même title ) sont listées et constituent une alternative textuelle valide, pourvu que leur support soit assuré.

La demande devrait plutôt être adressée au WCAG pour clarifier et homogénéiser sa position sur ce sujet. 

Aucune modification n’est envisagée.

**Comment by innoticFR** [comment]
Cela va effectivement dans le sens des dernières avancées du WCAG. C'est non conforme HTML, mais pas non conforme du point de vue du WCAG.
https://developer.paciellogroup.com/blog/2014/04/short-note-alt/

Il conviendrait toutefois d'apporter la précision figurant dans le F65:
> The `alt` attribute continues to be the preferred way to provide alternative text for images.

**Comment by innoticFR** [comment]
Statut suite au passage au RGAA 4.1

La note 6 qui a été ajoutée à l'entrée "Alternative textuelle (image)" mais ne concerne que l'attribut `title`.

Une adaptation de cette note permettrait de répondre à la problématique actuelle, mais en l'état les modifications apportées ne sont pas suffisantes

**Comment by acaodinsic** [comment]
Le ticket concerne la version RGAA 4.0.

**design-gouv** y donne une réponse qui est prise en compte dans la version RGAA 4.1

Je ferme donc ce ticket qui concerne la version RGAA 4.0

Néanmoins, si le point donne encore lieu à des questions dans la version RGAA 4.1, il convient d'ouvrir un nouveau ticket pour la version 4.1 avec éventuellement une référence liée à ce ticket, pour qu'elles puissent être analysées lors d'une amélioration ultérieure.
